GVRf/Framework/jni/engine/importer/assimp_importer.cpp

```diff
@@ -68,4 +71,176 @@ std::shared_ptr<Mesh> AssimpImporter::getMesh(int index) {
+                        if(mBitmap == NULL)
+                        {
+                            // Null bitmap for texture file for mesh
+                            // Applying default texture
+                            std::shared_ptr<Texture> texture(new BaseTexture(env, bitmap));
+                            material->setTexture("main_texture", texture);
+                        }
+                        else
+                        {
+                            // Back to native side from Java side
+                            // Applying actual texture
+                            std::shared_ptr<Texture> texture(new BaseTexture(env, mBitmap));
+                            material->setTexture("main_texture", texture);
+                        }
+                    }
+                }
+            }
+
+            // Set the material to the render data
+            renderData->set_material(material);
+
+            // Transformation
+            std::shared_ptr<Transform> transform(new Transform());
+            transform->set_owner_object(scnObject);
+
+            // Accumulated transformations of the node
+            aiMatrix4x4 aTransform = node->mTransformation * accumulatedTransform;
+            aiVector3t<float> scaling;
+            aiQuaterniont<float> rotation;
+            aiVector3t<float> position;
+            aTransform.Decompose(scaling, rotation, position);
+            int meshIndex = i;
+            transform->set_position(position.x, position.y, position.z);
+            transform->set_rotation(rotation.w, rotation.x, rotation.y, rotation.z);
+            transform->set_scale(scaling.x, scaling.y, scaling.z);
+            scnObject->attachTransform(scnObject, transform);
+            scnObject->attachRenderData(scnObject, renderData);
+
+            //Attaches the Scene Object to the Scene.
+            scnPtr->addSceneObject(scnObject);
+        } else {
+            //new RenderData class and perform all the operations
+        }
+    }
+
+    for(int i=0;i<node->mNumChildren;i++)
+    {
+        sceneRecursion(node->mChildren[i], aScene, scnPtr, env, obj, bitmap, accumulatedTransform);
+    }
+}
+
+
+std::shared_ptr<Scene> AssimpImporter::loadScene(JNIEnv* env, jobject obj, jobject bitmap)
+{
+    //  Scene
+    std::shared_ptr<Scene> scenePtr(new Scene());
+
+    // Left Camera
+    std::shared_ptr<Camera> leftCamera(new PerspectiveCamera());
+    leftCamera->set_render_mask(0x1);
+
+    //  Right Camera
+    std::shared_ptr<Camera> rightCamera(new PerspectiveCamera());
+    rightCamera->set_render_mask(0x2);
+
+    //  Left Camera object and its transform
+    std::shared_ptr<SceneObject> leftCameraObject(new SceneObject());
+    std::shared_ptr<Transform> transformLeftCameraObject(new Transform());
+    transformLeftCameraObject->set_owner_object(leftCameraObject);
+
+    float camera_separation_distance_ = 0.062f;
```

Comment:
Where does this 'magic number' come from? It really ought to be a symbolic constant, with a comment that explains the value and/or an expression that derives it from more fundamental constants.